libs/math/include/math/tensor_slice_iterator.hpp

```diff
@@ -203,8 +203,38 @@ class TensorSliceIterator
     std::swap(ranges_[a], ranges_[b]);
   }
 
-  // TODO: Name correctly
-  void MoveAxesToFront(SizeType const &a)
+  void MoveAxesToFront(std::vector<SizeType> const &a)
+  {
+    std::vector<TensorSliceIteratorRange> new_ranges;
+    new_ranges.reserve(ranges_.size());
+
+    // Insert axes at beginning
+    for (SizeType i = 0; i < a.size(); ++i)
+    {
+      new_ranges.push_back(ranges_[a[i]]);
+    }
+
+    for (SizeType i = 0; i < ranges_.size(); ++i)
+    {
+      // Search for axis
+      bool add_node = true;
```

Comment:
"add_node" does not make it clear what's going on here

Reply:
Why the hell I named it like that... :D